sktime/forecasting/base/adapters/_statsforecast.py

```diff
@@ -57,19 +54,92 @@ def _fit(self, y, X=None, fh=None):
         self : reference to self
         """
+        del fh  # avoid being detected as unused by ``vulture`` like tools
+
         self._forecaster = self._instantiate_model()
-        self._forecaster.fit(y.values, X.values if X is not None else X)
+
+        y_fit_input = y.to_numpy(copy=False)
+        X_fit_input = X.to_numpy(copy=False) if X is not None else X
+
+        self._forecaster.fit(y_fit_input, X=X_fit_input)
 
         return self
 
+    def _predict_in_or_out_of_sample(self, fh, fh_type, X=None, levels=None):
+        maximum_forecast_horizon = fh.to_relative(self.cutoff)[-1]
+
+        absolute_horizons = fh.to_absolute(self.cutoff).to_pandas()
+        horizon_positions = fh.to_indexer(self.cutoff)
+
+        level_arguments = None if levels is None else [100 * level for level in levels]
+
+        if fh_type == "in-sample":
+            predictions = self._forecaster.predict_in_sample(level=level_arguments)
+            point_predictions = predictions["fitted"]
+        elif fh_type == "out-of-sample":
+            predictions = self._forecaster.predict(
+                maximum_forecast_horizon, X=X, level=level_arguments
+            )
+            point_predictions = predictions["mean"]
+
+        if isinstance(point_predictions, pandas.Series):
+            point_predictions = point_predictions.to_numpy()
+
+        final_point_predictions = pandas.Series(
+            point_predictions[horizon_positions], index=absolute_horizons
+        )
+
+        if levels is None:
+            return final_point_predictions
+
+        interval_predictions_indices = pandas.MultiIndex.from_product(
+            [["Coverage"], levels, ["lower", "upper"]]
+        )
+        interval_predictions = pandas.DataFrame(
+            index=absolute_horizons, columns=interval_predictions_indices
+        )
+
+        if fh_type == "out-of-sample":
+            column_prefix = ""
+        elif fh_type == "in-sample":
+            column_prefix = "fitted-"
+
+        for level, level_argument in zip(levels, level_arguments):
+            lower_interval_predictions = predictions[
+                f"{column_prefix}lo-{level_argument}"
+            ]
+            if isinstance(lower_interval_predictions, pandas.Series):
+                lower_interval_predictions = lower_interval_predictions.to_numpy()
+
+            upper_interval_predictions = predictions[
+                f"{column_prefix}hi-{level_argument}"
+            ]
+            if isinstance(upper_interval_predictions, pandas.Series):
+                upper_interval_predictions = upper_interval_predictions.to_numpy()
+
+            interval_predictions[
+                ("Coverage", level, "lower")
+            ] = lower_interval_predictions[horizon_positions]
+            interval_predictions[
+                ("Coverage", level, "upper")
+            ] = upper_interval_predictions[horizon_positions]
+
+        return interval_predictions
+
+    def _split_horizon(self, fh):
```

Comment:
it occurs to me that almost always when we call one of the methods in there, we do this. Would it make sense to add this to `ForecastingHorizon`? Out of scope for this PR, a non-blocking comment.

Reply:
Yes.

In fact, getting end of horizon, splitting horizon in in-sample and out-of-sample, call separate method for both cases - all of this seems to be repeated for `pmdarima` and `statsforecast` (and also `tbats`?), and for both `predict` and `predict_interval`. A way to avoid these duplications will be useful.